packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,290 @@
+import Chai, { expect } from 'chai';
+import { sleep } from 'promise-assist';
+import { chaiRetryPlugin } from '../chai-retry-plugin/chai-retry-plugin';
+
+Chai.use(chaiRetryPlugin);
+
+const getFunctionToRetry = (func: (attempts: number) => unknown) => {
+    let callCount = 0;
+
+    const wrapperFunc = () => {
+        callCount++;
+        return func(callCount);
+    };
+
+    return {
+        funcToRetry: wrapperFunc,
+        getAttempts: () => {
+            return callCount;
+        },
+    };
+};
+
+describe('chai-retry-plugin', () => {
+    it('should retry a function that eventually succeeds', async () => {
+        const { funcToRetry, getAttempts } = getFunctionToRetry((attempts: number) => {
+            if (attempts < 3) {
+                throw new Error('Failed');
+            }
+            return 'Success';
+        });
+
+        await expect(funcToRetry).to.retry().to.equal('Success');
+        expect(getAttempts()).to.equal(3);
+    });
+
+    it('should retry a function that always fails and throw an error', async () => {
+        const { funcToRetry, getAttempts } = getFunctionToRetry(() => {
+            throw new Error('Failed');
+        });
+
+        try {
+            await expect(funcToRetry).to.retry({ retries: 3 });
+
+            throw new Error('This should not be called');
+        } catch (error) {
+            expect((error as Error).message).to.includes('Limit of 3 retries');
+            expect(getAttempts()).to.equal(3);
+        }
+    });
+
+    describe('options should work correctly:', () => {
+        it('timeout after the specified duration', async () => {
+            const funcToRetry = async () => {
+                await sleep(250);
+                return 'Success';
+            };
+
+            try {
+                await expect(funcToRetry).retry({ timeout: 700 }).to.equal('Failure');
+                throw new Error('This should not be called');
+            } catch (error: unknown) {
+                expect((error as Error).message).includes('Timed out after 700ms');
+            }
+        });
+
+        it('throw an error when retries limit exceeded', async () => {
```

Comment:
There already is a test for this - "should retry a function that always fails and throw an error"